projects/lendflare/convexBooster.js

```diff
@@ -0,0 +1,129 @@
+    // usdp
+    23: { lpToken: "0x7Eb40E450b9655f4B3cC4259BCC731c63ff55ae6", virtualBalance: "0xd8252a4EC16de33fAd1C69f2d3E80E9F7b8ceF12", coinId: 3, coinName: "tether", decimals: 6 },
+    // obtc
+    24: { lpToken: "0x2fE94ea3d5d4a175184081439753DE15AeF9d614", virtualBalance: "0x47A2eec0D7502Ba1FEc0cAe0eeafF64DCf28af60", coinId: 2, coinName: "wrapped-bitcoin", decimals: 8 },
+    // bbtc
+    25: { lpToken: "0x410e3E86ef427e30B9235497143881f717d93c2A", virtualBalance: "0xE1C0b5302E5a627141746ACB1025c43f9FE87ca8", coinId: 2, coinName: "wrapped-bitcoin", decimals: 8 },
+    // tbtc
+    26: { lpToken: "0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd", virtualBalance: "0x1ACc3ffb27c31C541695Ff75b21569CfaeDCA877", coinId: 2, coinName: "wrapped-bitcoin", decimals: 8 },
+    // pbtc
+    27: { lpToken: "0xDE5331AC4B3630f94853Ff322B66407e0D6331E8", virtualBalance: "0x0D66b49A68AffdDcDaDDdfE06CD6369307B2BA46", coinId: 2, coinName: "wrapped-bitcoin", decimals: 8 }
+}
+
+async function getTotalSupply(pools, timestamp, block, chainBlocks) {
+    const output = (await sdk.api.abi.multiCall({
+        block: chainBlocks.ethereum,
+        chain: "ethereum",
+        abi: 'erc20:totalSupply',
+        calls: Object.keys(pools).map((pid, _) => {
+            return {
+                target: pools[pid].virtualBalance
+            }
+        })
+    })).output.map((result, i) => {
+        for (pid of Object.keys(pools)) {
+            if (pools[pid].virtualBalance == result.input.target) {
+                pools[pid].totalSupply = (new BN(result.output));
+                pools[pid].totalSupplyString = result.output;
+            }
+        }
+    });
+
+    return pools;
+}
+
+async function calculateTokenAmount(pools, timestamp, block, chainBlocks) {
+    for (pid of Object.keys(pools)) {
+        await sdk.api.abi.call({
+            block: chainBlocks.ethereum,
+            chain: "ethereum",
+            abi: { "inputs": [{ "internalType": "uint256", "name": "_pid", "type": "uint256" }, { "internalType": "uint256", "name": "_tokens", "type": "uint256" }, { "internalType": "int128", "name": "_curveCoinId", "type": "int128" }], "name": "calculateTokenAmount", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" },
+            target: CONVEX_BOOSTER_PROXY,
+            params: [pid, pools[pid].totalSupplyString, pools[pid].coinId]
+        }).then(result => {
+            pools[pid].calculateTokenAmount = new BN(result.output);
+        }).catch(error => {
+            pools[pid].calculateTokenAmount = new BN(0);
+        });
+    }
+
+    return pools;
+}
+
+async function tvl(timestamp, block, chainBlocks) {
+    let tvl = new BN(0);
+
+    const pools = await getTotalSupply(convexPools, timestamp, block, chainBlocks).then(pools => {
+        return calculateTokenAmount(pools, timestamp, block, chainBlocks);
+    });
+    const prices = await utils.getPricesfromString().then(result => {
```

Comment:
sorry, missed this part. You probably need to pass the block data here as well, so you fetch the right prices